spec/unit/recipes/_configure_server_spec.rb

```diff
@@ -0,0 +1,354 @@
+      let(:expected_site) { { 'site' => 'site1' } }
+
+      it 'writes the server.conf file with the appropriate configs' do
+        expect(subject).to create_splunk_template('system/server.conf').with(stanzas: expected_attributes)
+        expect(subject.splunk_template('system/server.conf')).to notify('file[splunk-marker]').to(:touch)
+      end
+    end
+
+    context 'when the node is a cluster slave' do
+      let(:node_type) { :cluster_slave }
+      let(:expected_active_group) { 'Trial' }
+      let(:expected_site) { { 'site' => 'site1' } }
+      let(:replication_port) { { 'replication_ports' => { '8080' => {} } } }
+      let(:expected_clustering) { { 'clustering' => { 'master_uri' => 'https://cluster-master:8089', 'mode' => 'slave' } } }
+
+      it 'writes the server.conf file with the appropriate configs' do
+        expect(subject).to create_splunk_template('system/server.conf').with(stanzas: expected_attributes.merge(expected_replication_configs))
+        expect(subject.splunk_template('system/server.conf')).to notify('file[splunk-marker]').to(:touch)
+      end
+    end
+  end
+
+  context 'when singlesite cluster' do
+    let(:clusters) { ['cerner_splunk/singlesite_cluster'] }
+
+    context 'when the node is a cluster master' do
+      let(:node_type) { :cluster_master }
+      let(:expected_active_group) { 'Trial' }
+      let(:expected_clustering) { { 'clustering' => { 'replication_factor' => 2, 'search_factor' => 2, 'mode' => 'master' } } }
+
+      it 'writes the server.conf file with the appropriate configs' do
+        expect(subject).to create_splunk_template('system/server.conf').with(stanzas: expected_attributes)
+        expect(subject.splunk_template('system/server.conf')).to notify('file[splunk-marker]').to(:touch)
+      end
+    end
+
+    context 'when the node is a cluster slave' do
+      let(:node_type) { :cluster_slave }
+      let(:expected_active_group) { 'Trial' }
+      let(:replication_port) { { 'replication_ports' => { '8080' => {} } } }
+      let(:expected_clustering) { { 'clustering' => { 'master_uri' => 'https://cluster-master:8089', 'mode' => 'slave' } } }
+
+      it 'writes the server.conf file with the appropriate configs' do
+        expect(subject).to create_splunk_template('system/server.conf').with(stanzas: expected_attributes.merge(expected_replication_configs))
+        expect(subject.splunk_template('system/server.conf')).to notify('file[splunk-marker]').to(:touch)
+      end
+    end
+
+    context 'when the node is a forwarder' do
+      let(:node_type) { :forwarder }
+      let(:expected_site) { { 'site' => 'site1' } }
+
+      it 'writes the server.conf file with the appropriate configs' do
+        expect(subject).to create_splunk_template('system/server.conf').with(stanzas: expected_attributes)
+        expect(subject.splunk_template('system/server.conf')).to notify('file[splunk-marker]').to(:touch)
+      end
+    end
+  end
+
+  context 'when the node is a search head' do
```

Comment:
Can we add some additional detail to this context?  Same for line 237 since otherwise the contexts appear nearly identical.

Reply:
3819e106afcd180039e210ceb0cc8cc764a2ba7a